src/System.Linq.Parallel/tests/PlinqModesTests.cs

```diff
@@ -178,45 +148,18 @@ public static void WithExecutionMode_ArgumentNullException()
             Assert.Throws<ArgumentNullException>(() => ((ParallelQuery<int>)null).WithExecutionMode(ParallelExecutionMode.Default));
         }
 
-        private static void Enumerate<T>(this IEnumerable<T> e)
-        {
-            foreach (var x in e) { }
-        }
-
-        // A class that checks whether the Verify method got called from at least two threads.
-        // The first call to Verify() blocks. If another call to Verify() occurs prior to the timeout
-        // then we know that Verify() is getting called from multiple threads.
-        public class ParVerifier
+        /// <summary>Tracks all of the Tasks from which AddCurrent is called.</summary>
+        public sealed class UsedTaskTracker
         {
-            private int _counter = 0;
-            private bool _passed = false;
-            private const int TimeoutLimit = 30000;
+            private readonly ConcurrentDictionary<int, bool> _taskIdToUsageCount = new ConcurrentDictionary<int, bool>();
 
-            internal int Verify(int x)
+            internal int AddCurrent(int x)
             {
-                lock (this)
-                {
-                    _counter++;
-                    if (_counter == 1)
-                    {
-                        if (Monitor.Wait(this, TimeoutLimit))
-                        {
-                            _passed = true;
-                        }
-                    }
-                    else if (_counter == 2)
-                    {
-                        Monitor.Pulse(this);
-                    }
-                }
-
+                _taskIdToUsageCount.TryAdd(Task.CurrentId.Value, true);
```

Comment:
Do we expect that AddCurrent is called exactly once per task?  Do you want an assert that the return value of TryAdd is true?